uint32 arm_no
uint32 state_code
uint32 state_subcode
